### ISSUE TITLE
Question about detection map and softmax volume.

### ISSUE THREAD

**Issue by RenXiaoman** [DIAGNijmegen/picai_eval#26]
Hi, dear author.
I don't quite understand the difference between detection map and softmax. For example, my UNet output 2 classification results, and the normal process for obtaining labels is softmax ->argmax. So what is this detection map?

**Comment by joeranbosma** [comment]
Hi, the detection map is described here: https://pi-cai.grand-challenge.org/AI/

**Comment by RenXiaoman** [comment]
Thank you for your reply. 
============================================================================================
    for valid_data in valid_gen:
        idx += 1
        # if idx != 1:
        #     break
        try:
            valid_images = valid_data['data'].to(device)
            valid_labels = valid_data['seg']
        except Exception:
            valid_images = torch.from_numpy(valid_data['data']).to(device)  # ([4, 3, 20, 256, 256])
            valid_labels = valid_data['seg']  # (4, 1, 20, 256, 256)

            # test-time augmentation
            valid_images = [valid_images, torch.flip(valid_images, [4]).to(device)]


            # aggregate all validation predictions
            # gaussian blur to counteract checkerboard artifacts in
            # predictions from the use of transposed conv. in the U-Net
            # model(x)的输出是 [4, 2, 20, 256, 256]

        preds = [
            torch.sigmoid(model(x))[:, 1, ...].detach().cpu().numpy()
            for x in valid_images
        ]

        # revert horizontally flipped tta image
        preds[1] = np.flip(preds[1], [3])  # preds[(4, 20, 256, 256), (4, 20, 256, 256)]

        # gaussian blur to counteract checkerboard artifacts in
        # predictions from the use of transposed conv. in the U-Net
        preds = np.mean([   # [4, 20, 256, 256]
            gaussian_filter(x, sigma=1.5)
            for x in preds
        ], axis=0)

        # fig = plt.figure()
        # matshow3d(torch.tensor(preds[0]).unsqueeze(0), fig=fig, title="List of Volumes", frames_per_row=4, cmap="hot", )
        # plt.show()

        # extract lesion candidates
        preds = [  #  [4, 20, 256, 256]
            extract_lesion_candidates(x)[0]
            for x in preds
        ]



        for y_det, y_true in zip(preds, valid_labels):  #valid_labels shape is (4, 1, 20, 256, 256), preds shape is (4, 20, 256, 256)
            y_list, case_confidence, *_ = evaluate_case(
                y_det=y_det,
                y_true=y_true.squeeze(),
            )
            # print(case_confidence)

            # aggregate all validation evaluations
            lesion_results.append(y_list)
============================================================================================
Above is my code based on UNet in picai-baseline project.So this line
preds = [  #  [4, 20, 256, 256]
            extract_lesion_candidates(x)[0]
            for x in preds
        ]
The function "extract_lesion_candidates" is extracting the preds and then outputing the detection map, right?

**Comment by joeranbosma** [comment]
Yes, `extract_lesion_candidates` indeed extracts the lesion candidates from the softmax volume, turning it into the detection map.
